pkg/docker/docker.go

```diff
@@ -864,54 +1073,60 @@ func (d *stiDocker) GetImageID(name string) (string, error) {
 // CommitContainer commits a container to an image with a specific tag.
 // The new image ID is returned
 func (d *stiDocker) CommitContainer(opts CommitContainerOptions) (string, error) {
-	repository, tag := docker.ParseRepositoryTag(opts.Repository)
-	dockerOpts := docker.CommitContainerOptions{
-		Container:  opts.ContainerID,
-		Repository: repository,
-		Tag:        tag,
+	ctx, cancel := getDefaultContext(DefaultDockerTimeout)
+	defer cancel()
+	dockerOpts := dockertypes.ContainerCommitOptions{
+		Reference: opts.Repository,
 	}
 	if opts.Command != nil || opts.Entrypoint != nil {
-		config := docker.Config{
+		config := dockercontainer.Config{
 			Cmd:        opts.Command,
 			Entrypoint: opts.Entrypoint,
 			Env:        opts.Env,
 			Labels:     opts.Labels,
 			User:       opts.User,
 		}
-		dockerOpts.Run = &config
+		dockerOpts.Config = &config
 		glog.V(2).Infof("Committing container with dockerOpts: %+v, config: %+v", dockerOpts, config)
 	}
 
-	image, err := d.client.CommitContainer(dockerOpts)
-	if err == nil && image != nil {
-		return image.ID, nil
+	resp, err := d.client.ContainerCommit(ctx, opts.ContainerID, dockerOpts)
+	if err == nil {
+		return resp.ID, nil
 	}
 	return "", err
 }
 
 // RemoveImage removes the image with specified ID
 func (d *stiDocker) RemoveImage(imageID string) error {
-	return d.client.RemoveImage(imageID)
+	_, err := d.kubeDockerClient.RemoveImage(imageID, dockertypes.ImageRemoveOptions{})
+	return err
 }
 
 // BuildImage builds the image according to specified options
 func (d *stiDocker) BuildImage(opts BuildImageOptions) error {
-	dockerOpts := docker.BuildImageOptions{
-		Name:                opts.Name,
-		NoCache:             true,
-		SuppressOutput:      false,
-		RmTmpContainer:      true,
-		ForceRmTmpContainer: true,
-		InputStream:         opts.Stdin,
-		OutputStream:        opts.Stdout,
+	dockerOpts := dockertypes.ImageBuildOptions{
+		Tags:           []string{opts.Name},
+		NoCache:        true,
+		SuppressOutput: false,
 	}
 	if opts.CGroupLimits != nil {
 		dockerOpts.Memory = opts.CGroupLimits.MemoryLimitBytes
-		dockerOpts.Memswap = opts.CGroupLimits.MemorySwap
+		dockerOpts.MemorySwap = opts.CGroupLimits.MemorySwap
 		dockerOpts.CPUShares = opts.CGroupLimits.CPUShares
 		dockerOpts.CPUPeriod = opts.CGroupLimits.CPUPeriod
 		dockerOpts.CPUQuota = opts.CGroupLimits.CPUQuota
 	}
 	glog.V(2).Infof("Building container using config: %+v", dockerOpts)
-	return d.client.BuildImage(dockerOpts)
+	ctx, cancel := getDefaultContext(((1<<63 - 1) * time.Nanosecond)) // infinite duration ... go does not expost max duration constant
+	defer cancel()
+	resp, err := d.client.ImageBuild(ctx, opts.Stdin, dockerOpts)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	// since can't pass in output stream to engine-api, need to copy contents of
+	// the output stream they create into our output stream
+	_, err = io.Copy(opts.Stdout, resp.Body)
```

Comment:
comparing go-dockerclient and engine-api, it is essentially the same code, i.e the call to to a copy on the same thread of execution....however, go-dockerclient, when not raw terminal, has demux version of copy which buffers the processing of the http resp body into stdout/stderr.  Perhaps that gives the illusion of progressive updating of the build logs.

I'll have to run some experiments, comparing to two forms, to confirm the similarity, or uncover the difference.